go/pkg/services/logcollector/server.go

```diff
@@ -785,7 +798,16 @@ func (s *Server) getLogFilePath(ctx context.Context, runUID, projectName string)
 		return filePath.(string), nil
 	}
 
-	logFilePath := ""
+	// get project mutex or create one
+	projectMutex, _ := s.readDirentProjectNameSyncMap.LoadOrStore(projectName, &sync.Mutex{})
+
+	// lock project mutex, we want only one project dir to be read at a time
+	projectMutex.(*sync.Mutex).Lock()
```

Comment:
type assertion might fail?

Reply:
panic to clean memory in case some funky stuff happen (which shouldnot)